### PR TITLE
Add more robust test of io

### PR DESCRIPTION
**Pull request by znicholls**
# Pull request

Please confirm that this pull request has done the following:

- [x] Tests added
- [x] Documentation added (where applicable)
- [x] Example added (either to an existing notebook or as a new notebook, where applicable) (N/A, all internal)
- [x] Description in ``CHANGELOG.rst`` added

## Adding to CHANGELOG.rst

Please add a single line in the changelog notes similar to one of the following:

```
- (`#XX <http://link-to-pr.com>`_) Added feature which does something
- (`#XX <http://link-to-pr.com>`_) Fixed bug identified in (`#XX <http://link-to-issue.com>`_)
```

### CODE REVIEW

**Comment by znicholls** [comment]
@lewisjared can you take a look when you've got a minute? Let's not merge until @rgieseke gives the approval though

File: pymagicc/io.py [hunk 1]
```diff
@@ -807,17 +807,22 @@ def read(self):
...
-            elif metadata["unit"] == "metric tons":
-                emms = True
-            else:
-                error_msg = "I do not recognise the unit, {}, in file{}".format(
-                    metadata["unit"], self.filepath
-                )
-                raise ValueError(error_msg)
-        else:
+
+        unit_keys = ["Unit", "unit"]
```
Comment (lewisjared) [inline pymagicc/io.py]: Why not perform a case insensitive lookup (using tolower())?

File: pymagicc/io.py [hunk 2]
```diff
@@ -872,13 +877,14 @@ def _read_header(self):
...
+        data_block_header_line = self._stream.readline().replace("Years", "").strip()
         variables = []
         col_width = 10
-        for n in np.arange(0, len(data_block_header_line), col_width):
-            pos = int(n)
-            variable = data_block_header_line[pos : pos + col_width].strip()
+
+        for w in range(0, len(data_block_header_line), col_width):
+            variable = data_block_header_line[w : w + col_width].strip()
             if variable != "Years":
```
Comment (lewisjared) [inline pymagicc/io.py]: You have already stripped out "Years" in L882

**Comment by znicholls** [comment]
> Are any of these expected files actually run through magicc to be sure that they work?

They are now (except the OT file because I have no idea how that's used and hence how to test it...)

**Comment by znicholls** [comment]
@rgieseke if you have time would be great to get your thoughts

File: pymagicc/io.py [hunk 3]
```diff
@@ -807,20 +807,27 @@ def read(self):
...
-                error_msg = "I do not recognise the unit, {}, in file{}".format(
-                    metadata["unit"], self.filepath
+
+        unit_keys = [k for k in metadata.keys() if k.lower() == "unit"]
+        units = [metadata.pop(k) for k in unit_keys]
+        if not units:
+            # have to assume global emissions in tons
+            emms = True
+        elif len(units) != 1:
+            import pdb
```
Comment (rgieseke) [inline pymagicc/io.py]: Looks super useful on one hand, but could be very irritating to users not used to Python debugging ...

Comment (rgieseke) [reply pymagicc/io.py]: Or is this a left-over?

**Comment by rgieseke** [comment]
Excellent documentation of pit-falls, very useful to have this written down here!
Only question from my side the debug statement ...

File: pymagicc/io.py (hunk 3 above)
Comment (znicholls) [reply pymagicc/io.py]: left over, good pick up

**Comment by znicholls** [comment]
@lewisjared over to you to merge when you're happy then I think

**Pull request merged**